src/components/ui/loadingSpinner.tsx

```diff
@@ -0,0 +1,26 @@
+/**
+ *
+ * A Loading spinner component to show while data is updating
+ *
+ * @example
+ * layout.tsx
+ *
+ * <html lang="en">
+ *   <body className={cn('flex justify-center items-start w-screen min-h-screen bg-primary-lightest', roboto.className,)}>
+ *     <LoadingSpinner />
+ *     <main className={'max-w-screen-sm min-h-screen w-full h-full bg-white px-6'}>
+ *       {children}
+ *     </main>
+ *     <Toaster />
+ *   </body>
+ * </html>
+ *
+ */
+
+export const LoadingSpinner = () => {
```

Comment:
Could you add isLoading prop for displaying the loading spinner separately?

▼Example
```export const LoadingSpinner = (isLoading: boolean) => {
  return (
    <>
      {isLoading && (
        <div className="absolute left-0 top-0 z-50 bg-black/50 w-full h-full flex justify-center items-center">
          <div className=" border-gray-200 h-12 w-12 animate-spin rounded-full border-4 border-t-primary" />
        </div>
      )}
    </>
  );
};
```

Reply:
Thank you for your comments!
Following Akio's discussion, we made these changes:
- Added `useLockBodyScroll.tsx` in the hooks folder and implemented the logic there.
- Updated `loadingSpinner.tsx` to call useLockBodyScroll and use the isLoading prop to control the spinner's visibility.